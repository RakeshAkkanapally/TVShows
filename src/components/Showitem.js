import React from "react";
import {
  Card,
  CardContent,
  CardMedia,
  makeStyles,
  Typography,
} from "@material-ui/core";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(2),
    maxWidth: theme.spacing(30),
    alignItems: "center",
  },
  card: {
    width: theme.spacing(30),
    height: theme.spacing(45),
  },
  header: {
    maxHeight: theme.spacing(1),
    fontSize: theme.spacing(1),
  },
  media: {
    margin: "0 auto 0",
    width: theme.spacing(33),
    height: theme.spacing(45),
  },
  content: {
    margin: "0 auto 0",
    textAlign: "center",
    fontSize: theme.spacing(1.5),
  },
}));

const ShowItem = (props) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Card className={classes.card}>
        {props.url ? (
          <NavLink
            to={
              props.url === "show"
                ? `/show/${props.id}`
                : `/show/${props.id}/episodes/${props.snum}/${props.epnum}`
            }
          >
            <CardMedia
              className={classes.media}
              image={getImage(props.image, props.imageType)}
            ></CardMedia>
          </NavLink>
        ) : (
          <CardMedia
            className={classes.media}
            image={getImage(props.image, props.imageType)}
          ></CardMedia>
        )}
        <CardContent>
          {props.rows &&
            props.rows.map((row) => (
              <Typography className={classes.content} key={row.name}>
                <strong>{row.name + ":"} </strong>
                {row.value ? `${row.value}` : "NA"}
              </Typography>
            ))}
        </CardContent>
      </Card>
    </div>
  );
};

function getImage(image, type) {
  return image
    ? image[type]
    : "https://www.publicdomainpictures.net/pictures/280000/velka/not-found-image-15383864787lu.jpg";
  
  }

ShowItem.propTypes = {
  name: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  num: PropTypes.number,
  image: PropTypes.object,
  imageType: PropTypes.string,
  medium: PropTypes.string,
  rating: PropTypes.object,
  average: PropTypes.string,
  genres: PropTypes.array,
  episodeOrder: PropTypes.number,
  url: PropTypes.string,
  rows: PropTypes.array,
  snum: PropTypes.number,
  epnum: PropTypes.number,
};

export default ShowItem;
