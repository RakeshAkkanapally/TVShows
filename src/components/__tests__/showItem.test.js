import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Showitem from "../ShowItem";

describe("verifying ShowItem component", () => {
  it("ShowItem renders correctly", () => {
    let props = {
      id: 1,
      name: "Under the Dome",
      genres: ["Drama", "Science-Fiction", "Thriller"],
      rating: {
        average: 6.6,
      },
      image: {
        medium:
          "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
      },
      rows: [{ name: "Genre", value: ["Drama", "Action"] }],
      snum: 9,
      epnum: 9,
    };

    const tree = renderer
      .create(
        <BrowserRouter>
          <Showitem {...props} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when few props are not passed", () => {
    let props = {
      id: 1,
      url: "show",
      name: "Under the Dome",
      rating: {},
      rows: [{ name: "Genre" }],
      image: {
        medium:
          "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
      },
    };
    const tree = renderer
      .create(
        <BrowserRouter>
          <Showitem {...props} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when few props are not passed", () => {
    let props = {
      id: 1,
      url: "show",
      name: "Under the Dome",
      rating: {},
      rows: [{ name: "Genre" }],
    };
    const tree = renderer
      .create(
        <BrowserRouter>
          <Showitem {...props} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when few props are not passed.", () => {
    let props = {
      id: 1,
      url: "NA",
      rating: {},
      rows: [{ name: "Genre" }],
    };
    const tree = renderer
      .create(
        <BrowserRouter>
          <Showitem {...props} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when few props are not passed.", () => {
    let props = {
      id: 1,
      name: "Under the Dome",
      genres: ["Drama", "Science-Fiction", "Thriller"],
      rating: {
        average: 6.6,
      },
      rows: [{ name: "Genre", value: ["Drama", "Action"] }],
      snum: 9,
      epnum: 9,
    };
    const tree = renderer
      .create(
        <BrowserRouter>
          <Showitem {...props} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
