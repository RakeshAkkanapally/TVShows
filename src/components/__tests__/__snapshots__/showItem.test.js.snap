// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verifying ShowItem component ShowItem renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root makeStyles-header-3"
    >
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <div
      className="MuiCardMedia-root makeStyles-media-4"
      title="Under the Dome"
    >
       
    </div>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-content-5 MuiTypography-body1"
      >
        <strong>
          Genre:
           
        </strong>
        Drama,Action
      </p>
    </div>
  </div>
</div>
`;

exports[`verifying ShowItem component renders correctly when few props are not passed 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root makeStyles-header-3"
    >
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <a
      aria-current={null}
      href="/show/1"
      onClick={[Function]}
    >
      <div
        className="MuiCardMedia-root makeStyles-media-4"
        style={
          Object {
            "backgroundImage": "url(\\"https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\")",
          }
        }
        title="Under the Dome"
      >
         
      </div>
    </a>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-content-5 MuiTypography-body1"
      >
        <strong>
          Genre:
           
        </strong>
        NA
      </p>
    </div>
  </div>
</div>
`;

exports[`verifying ShowItem component renders correctly when few props are not passed 2`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root makeStyles-header-3"
    >
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <a
      aria-current={null}
      href="/show/1"
      onClick={[Function]}
    >
      <div
        className="MuiCardMedia-root makeStyles-media-4"
        style={
          Object {
            "backgroundImage": "url(\\"https://www.publicdomainpictures.net/pictures/280000/velka/not-found-image-15383864787lu.jpg\\")",
          }
        }
        title="Under the Dome"
      >
         
      </div>
    </a>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-content-5 MuiTypography-body1"
      >
        <strong>
          Genre:
           
        </strong>
        NA
      </p>
    </div>
  </div>
</div>
`;

exports[`verifying ShowItem component renders correctly when few props are not passed. 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root makeStyles-header-3"
    >
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <a
      aria-current={null}
      href="/show/1/episodes/undefined/undefined"
      onClick={[Function]}
    >
      <div
        className="MuiCardMedia-root makeStyles-media-4"
        style={
          Object {
            "backgroundImage": "url(\\"https://www.publicdomainpictures.net/pictures/280000/velka/not-found-image-15383864787lu.jpg\\")",
          }
        }
      >
         
      </div>
    </a>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-content-5 MuiTypography-body1"
      >
        <strong>
          Genre:
           
        </strong>
        NA
      </p>
    </div>
  </div>
</div>
`;

exports[`verifying ShowItem component renders correctly when few props are not passed. 2`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root makeStyles-header-3"
    >
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <div
      className="MuiCardMedia-root makeStyles-media-4"
      style={
        Object {
          "backgroundImage": "url(\\"https://www.publicdomainpictures.net/pictures/280000/velka/not-found-image-15383864787lu.jpg\\")",
        }
      }
      title="Under the Dome"
    >
       
    </div>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-content-5 MuiTypography-body1"
      >
        <strong>
          Genre:
           
        </strong>
        Drama,Action
      </p>
    </div>
  </div>
</div>
`;
